### PR TITLE
Fix: MVT update min/maXZoom props

### PR DESCRIPTION
**Pull request by alasarr**
Changes on `minZoom` or `maxZoom` props didn't work at MVTLayer. Reported at https://github.com/visgl/deck.gl/issues/6023

### CODE REVIEW

File: modules/geo-layers/src/mvt-layer/mvt-layer.js [hunk 1]
```diff
@@ -70,13 +72,13 @@ export default class MVTLayer extends TileLayer {
     if (tileJSON) {
       data = tileJSON.tiles;
 
-      if (Number.isFinite(tileJSON.minzoom) && tileJSON.minzoom > minZoom) {
+      if (Number.isFinite(tileJSON.minzoom) && tileJSON.minzoom > this.props.minZoom) {
```
Comment (Pessimistress) [inline modules/geo-layers/src/mvt-layer/mvt-layer.js:75]: This is still depending on a prop that could later change. This chunk pausing should be performed in getTilesetOptions

Comment (alasarr) [reply modules/geo-layers/src/mvt-layer/mvt-layer.js]: Good catch, thanks, another approach implemented.

**Review by Pessimistress — APPROVED** [review]

File: modules/geo-layers/src/mvt-layer/mvt-layer.js [hunk 2]
```diff
@@ -69,21 +69,30 @@ export default class MVTLayer extends TileLayer {
...
         Number.isFinite(tileJSON.maxzoom) &&
-        (!Number.isFinite(maxZoom) || tileJSON.maxzoom < maxZoom)
+        (!Number.isFinite(props.maxZoom) || tileJSON.maxzoom < props.maxZoom)
       ) {
-        maxZoom = tileJSON.maxzoom;
+        opts.maxZoom = tileJSON.maxzoom;
       }
     }
-
-    this.setState({data, tileJSON, minZoom, maxZoom});
```
Comment (Pessimistress) [inline modules/geo-layers/src/mvt-layer/mvt-layer.js:85]: Can we remove this now?

https://github.com/visgl/deck.gl/blob/5338f9abbcb253c44fcc3464676591255724431a/modules/geo-layers/src/tile-layer/tile-layer.js#L85-L86

Comment (alasarr) [reply modules/geo-layers/src/mvt-layer/mvt-layer.js:85]: As we've redefined `_getTilesetOptions`, maxZoom and minZoom are directly returned no need to be in the state. 

I'm thinking now that this two lines https://github.com/visgl/deck.gl/blob/5338f9abbcb253c44fcc3464676591255724431a/modules/geo-layers/src/tile-layer/tile-layer.js#L85-L86 can be removed now.

File: modules/geo-layers/src/tile-layer/tile-layer.js [hunk 3]
```diff
@@ -82,14 +82,14 @@ export default class TileLayer extends CompositeLayer {
   }
 
   _getTilesetOptions(props) {
-    const maxZoom = Number.isFinite(this.state.maxZoom) ? this.state.maxZoom : props.maxZoom;
```
Comment (alasarr) [inline modules/geo-layers/src/tile-layer/tile-layer.js:85]: @Pessimistress do you see something wrong with this? I've tested with mvtlayer and it works.

Comment (Pessimistress) [reply modules/geo-layers/src/tile-layer/tile-layer.js:85]: Perfect, I think this is the right thing to do (TileLayer should not reference a state that itself doesn't define).

**Pull request merged**